compiler-core/src/error.rs

```diff
@@ -2041,6 +2041,27 @@ value, or use `assert` rather than `let`."
                         extra_labels: vec![],
                     }),
                 },
+
+                TypeError::TypeCycle { location, cycle } =>  {
+                    let mut text = "The definition of that type forms a cycle:\n".into();
+                    write_cycle(&mut text, cycle);
+                    text.push_str("Gleam doesn't support type cycles like these, please break the cycle to continue.");
```

Comment:
```suggestion
                    text.push_str("If we tried to compile this recursive type it would expand
forever in a loop, and we'd never get the final type.");
```

I think it's nice to explain why it doesn't work when possible, so the programmer thinks we're not locking away functionality for no reason.

Reply:
Makes sense